system/Model.php

```diff
@@ -1634,6 +1634,31 @@ public function __get(string $name)
 		return null;
 	}
 
+	/**
+	 * Checks for the existence of properties across this model, builder, and db connection.
+	 *
+	 * @param string $name
+	 *
+	 * @return bool
+	 */
+	public function __isset(string $name): bool
+	{
+		if (in_array($name, ['primaryKey', 'table', 'returnType', 'DBGroup']))
+		{
+			return true;
+		}
+		elseif (isset($this->db->$name))
+		{
+			return true;
+		}
+		elseif (isset($this->builder()->$name))
+		{
+			return true;
+		}
+
```

Comment:
One thing this doesn't account for is any custom properties a user might have put in their model. I think this needs a `property_exists()` check. Maybe instead of the first check? Unsure.

Reply:
Yes I think you're right. Added via 4157f2965